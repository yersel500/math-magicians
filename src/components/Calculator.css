.mayorContainer {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  padding: 1rem;
  height: 100vh;
}

.calContainer {
  width: 50%;
  height: 50%;
}

.result-container {
  display: flex;
  justify-content: flex-end;
  align-items: flex-end;
  background-color: rgb(165, 163, 163);
}

.result {
  color: #fff;
  padding: 1rem;
}

.keyContainer {
  display: grid;
  justify-items: stretch;
  align-items: stretch;
  align-content: stretch;
  grid-template-columns: 1fr 1fr 1fr 1fr;
  grid-template-rows: auto auto auto auto auto;
  height: 100%;
}

.my-0 {
  grid-column-start: 1;
  grid-column-end: 3;
}

.operator {
  background-color: rgb(243, 133, 105);
}
