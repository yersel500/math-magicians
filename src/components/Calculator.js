import './Calculator.css';
import PropTypes from 'prop-types';
import React from 'react';
import calculate from '../logic/calculate';

const ShowResult = (props) => {
  const { result } = props;
  return (
    <div className="result-container">
      <p className="result">{result}</p>
    </div>
  );
};

ShowResult.defaultProps = { result: 0 };
ShowResult.propTypes = { result: PropTypes.number };

const KeyCalc = (props) => {
  const { eachKey, myClass } = props;
  return (
    <input type="button" className={`eachKey my-${eachKey} ${myClass}`} value={eachKey} onClick={handleOperation} />
  );
};

KeyCalc.defaultProps = {
  eachKey: '0',
  myClass: 'general',
};
KeyCalc.propTypes = {
  eachKey: PropTypes.string,
  myClass: PropTypes.string,
};

export default class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: null,
      next: null,
      operation: null,
    }
    this.myKeys = [
      {
        eachKey: 'AC',
        myClass: 'general',
      },
      {
        eachKey: '+/-',
        myClass: 'general',
      },
      {
        eachKey: '%',
        myClass: 'general',
      },
      {
        eachKey: 'รท',
        myClass: 'operator',
      },
      {
        eachKey: '7',
        myClass: 'number',
      },
      {
        eachKey: '8',
        myClass: 'number',
      },
      {
        eachKey: '9',
        myClass: 'number',
      },
      {
        eachKey: 'x',
        myClass: 'operator',
      },
      {
        eachKey: '4',
        myClass: 'number',
      },
      {
        eachKey: '5',
        myClass: 'number',
      },
      {
        eachKey: '6',
        myClass: 'number',
      },
      {
        eachKey: '-',
        myClass: 'operator',

      },
      {
        eachKey: '1',
        myClass: 'number',

      },
      {
        eachKey: '2',
        myClass: 'number',

      },
      {
        eachKey: '3',
        myClass: 'number',

      },
      {
        eachKey: '+',
        myClass: 'operator',

      },
      {
        eachKey: '0',
        myClass: 'number',

      },
      {
        eachKey: '.',
        myClass: 'general',

      },
      {
        eachKey: '=',
        myClass: 'operator',

      },
    ];
  }

  render() {
    return (
      <div className="mayorContainer">
        <div className="calContainer">
          <ShowResult />
          <div className="keyContainer">
            {this.myKeys.map((element) => (
              <KeyCalc eachKey={element.eachKey} myClass={element.myClass} key={element.eachKey} />
            ))}
          </div>
        </div>
      </div>
    );
  }
}
