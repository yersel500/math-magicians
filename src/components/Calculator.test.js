import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Calculator from './Calculator';

it('The calculator renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

describe('Calculator', () => {
  test('renders App component', () => {
    render(<Calculator />);
    screen.debug();
  });
});
