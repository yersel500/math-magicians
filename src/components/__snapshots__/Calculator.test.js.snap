// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The calculator renders correctly 1`] = `
<div
  className="mayorContainer"
>
  <div>
    <h2>
      Let's do some math!
    </h2>
  </div>
  <div
    className="calContainer"
  >
    <div
      className="result-container"
    >
      <p
        className="result"
      >
        0
      </p>
    </div>
    <div
      className="keyContainer"
    >
      <input
        className="eachKey my-AC general"
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        className="eachKey my-+/- general"
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        className="eachKey my-% general"
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="eachKey my-รท operator"
        onClick={[Function]}
        type="button"
        value="รท"
      />
      <input
        className="eachKey my-7 number"
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        className="eachKey my-8 number"
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        className="eachKey my-9 number"
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="eachKey my-x operator"
        onClick={[Function]}
        type="button"
        value="x"
      />
      <input
        className="eachKey my-4 number"
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        className="eachKey my-5 number"
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        className="eachKey my-6 number"
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="eachKey my-- operator"
        onClick={[Function]}
        type="button"
        value="-"
      />
      <input
        className="eachKey my-1 number"
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        className="eachKey my-2 number"
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        className="eachKey my-3 number"
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="eachKey my-+ operator"
        onClick={[Function]}
        type="button"
        value="+"
      />
      <input
        className="eachKey my-0 number"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        className="eachKey my-. general"
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="eachKey my-= operator"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </div>
</div>
`;
